src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -3919,6 +3966,82 @@ impl<'a> TypeChecker<'a> {
         }
     }
 
+    fn rewritable_variant_functions() -> &'static [Ascii<&'static str>] {
+        static VARIANT_FUNCTIONS: &[Ascii<&'static str>] = &[
+            Ascii::new("get"),
+            Ascii::new("get_string"),
+            Ascii::new("get_by_keypath"),
+            Ascii::new("get_by_keypath_string"),
```

Comment:
`get_string` and `get_by_keypath_string` is equivalent to `get` and `get_by_keypath` with cast to string data type, need also modify the virtual column type.